### PR TITLE
Support Memory Hotplug with Hugepages

### PR DESCRIPTION
**Pull request by acardace**
### What this PR does
Before this PR:
A VM configured with memory hotplug and hugepages does not start

After this PR:
A VM configured with memory hotplug and hugepages starts correctly

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
Support Memory Hotplug with Hugepages
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/converter/converter.go [hunk 1]
```diff
@@ -86,7 +86,10 @@ const (
 	// must be a power of 2 and at least equal
 	// to the size of a transparent hugepage (2MiB on x84_64).
 	// Recommended value by QEMU is 2MiB
-	MemoryHotplugBlockAlignmentBytes = 2097152
+	MemoryHotplugBlockAlignmentBytes = 0x200000
```
Comment (jean-edouard) [inline pkg/virt-launcher/virtwrap/converter/converter.go:89]: :pray:

File: pkg/virt-launcher/virtwrap/converter/converter.go [hunk 2]
```diff
@@ -1224,12 +1226,19 @@ func setupDomainMemory(vmi *v1.VirtualMachineInstance, domain *api.Domain) error
...
+		blockAlignment = MemoryHotplug1GHugePagesBlockAlignmentBytes
+	}
+
 	domain.Spec.Devices.Memory = &api.MemoryDevice{
 		Model: "virtio-mem",
 		Target: &api.MemoryTarget{
 			Size:  pluggableMemorySize,
 			Node:  "0",
-			Block: api.Memory{Unit: "b", Value: MemoryHotplugBlockAlignmentBytes},
+			Block: api.Memory{Unit: "b", Value: uint64(blockAlignment)},
```
Comment (jean-edouard) [inline pkg/virt-launcher/virtwrap/converter/converter.go:1241]: Nit: can't we just declare the consts as uint64?

**Review by jean-edouard — COMMENTED** [review]
/approve

File: pkg/virt-launcher/virtwrap/converter/converter.go (hunk 2 above)
Comment (acardace) [reply pkg/virt-launcher/virtwrap/converter/converter.go:1241]: Yes but then it gets used in other places, see https://github.com/search?q=repo%3Akubevirt%2Fkubevirt%20MemoryHotplugBlockAlignmentBytes&type=code.

So I preferred to have less casts globally :)

**Review by fossedihelm — APPROVED** [review]

**Pull request merged**